### PR TITLE
Align transient dependencies, expose through re-exports

### PR DESCRIPTION
**Pull request by Schwusch**
This is an alternative to https://github.com/agilord/aws_client/pull/106

### CODE REVIEW

File: generator/lib/builders/library_builder.dart [hunk 1]
```diff
@@ -44,9 +44,6 @@ String buildService(Api api) {
 import 'dart:convert';
 import 'dart:typed_data';
 
-import 'package:json_annotation/json_annotation.dart';
-import 'package:meta/meta.dart' as _meta;
-
 import 'package:shared_aws_api/shared.dart' as shared;
 import 'package:shared_aws_api/shared.dart' show Uint8ListConverter, Uint8ListListConverter;
```
Comment (isoos) [inline generator/lib/builders/library_builder.dart]: I'm wondering if we still need this, but the rest looks good.

Comment (Schwusch) [reply generator/lib/builders/library_builder.dart]: What are you referring to? The lines I removed or the ones that are left?

Comment (isoos) [reply generator/lib/builders/library_builder.dart]: The unprefixed `show Uint8ListConverter, Uint8ListListConverter`

Comment (isoos) [reply generator/lib/builders/library_builder.dart]: The reason being that now json-annotation will be behind the shared prefix too, and maybe that could fix the build_runner issue. Also a side not: we should probably rename it to `_shared` (or `_s`).

**Pull request merged**